### PR TITLE
CEM-2547

### PR DESCRIPTION
**Pull request by Zhuk314**
Create an Oval Table and Layout the Chairs around the table.

### CODE REVIEW

**Comment by Zhuk314** [comment]
12 chairs
3 chairs
9 chairs

File: src/Containers/OvalTable/OvalTable.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+            isSeated: true,
+            occupiedBy: 'Dean',
+            isVisible: true,
+            isRound: true,
+            relativeSize: 0.5,
+            tableUse: 'TableForManagement',
+            chairIndex: 0,
+            tableIndex: 0,
+            selectedIndex: 3,
+            onChairClick: handleOnChairClick,
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.stories.tsx:71]: Use the storybook action event

File: src/Containers/OvalTable/OvalTable.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    return `
+        border-width: ${relativeSize * BASE_TABLE_BORDER_WIDTH}em;
+        margin: ${relativeSize * BASE_TABLE_BODY_MARGIN}em;
+        width: ${relativeSize * BASE_TABLE_WIDTH}em;
+        height: ${relativeSize * BASE_TABLE_HEIGHT}em;
+        `;
+  }}
+    
+  position: relative;
+  background-color: gray;
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.tsx:180]: Use theme

File: src/Containers/OvalTable/OvalTable.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        partyName = 'Null',
+        relativeSize = 1.0,
+        tableUse = 'TableForManagement',
+        tableIndex = 0,
+        selectedIndex = -1,
+        onChairClick,
+
+        ...props
+    }) => {
+    const getChairs: getChairsType = () =>
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.tsx:54]: Missing JSDOC

**Review by ralph-dev — CHANGES_REQUESTED** [review]

File: src/Containers/OvalTable/OvalTable.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+const Template: Story<IOvalTable> = (args) => <OvalTable {...args} />;
+
+/**
+ * Prints the Selected Child index to the console when Chair is clicked
+ * @param tableIndex
+ * @param chairIndex
+ * @param selectedTableIndex
+ */
+const handleOnChairClick = (
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.stories.tsx:19]: Remove this and replace the console log with storybook actions so it emits the correct event.

action("Log Data")

**Review by ralph-dev — CHANGES_REQUESTED** [review]

File: src/Containers/OvalTable/OvalTable.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    let quadrantBasedAngle = CHAIR_ROTATION_ANGLE % STRAIGHT_ANGLE;
+
+    // In odd quadrants (angle > 90) increasing angle increase radius,
+    // so the angle can be proportionally inverted. For even quadrants, the angle remain the same.
+    if(quadrantBasedAngle > RIGHT_ANGLE){
+        quadrantBasedAngle = STRAIGHT_ANGLE - quadrantBasedAngle
+    }
+
+    // Calculate the coefficient which may vary from 0.7 to 1, based on angle of the chair.
+    const translationCoefficient = 1 - (HEIGHT_TO_WIDTH_DIFFERENCE * (quadrantBasedAngle/RIGHT_ANGLE));
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.tsx:145]: This 1 is a magic number

File: src/Containers/OvalTable/OvalTable.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    numOfChairs: number;
+    relativeSize: number;
+}
+
+const ChairWrapper = styled.div<IChairWrapper>`
+  --chairDiameter: ${({ relativeSize }) => {
+    const BASE_CHAIR_SIZE = 6.5;
+    return BASE_CHAIR_SIZE * relativeSize;
+  }}em; /* chair size */
+
```
Comment (ralph-dev) [inline src/Containers/OvalTable/OvalTable.tsx:193]: This is hardcoded, we should check if this works with Jadas Chair she built

**Review by ralph-dev — CHANGES_REQUESTED** [review]

**Review by ralph-dev — APPROVED** [review]
Lgtm assuming comments have been resolved.

**Pull request merged**